src/generic-container/generic-container.ts

```diff
@@ -206,12 +206,12 @@ export class GenericContainer implements TestContainer {
     return this;
   }
 
-  public withExposedPorts(...ports: Port[]): this {
+  public withExposedPorts(...ports: PortWithOptionalBinding[]): this {
```

Comment:
the base interface `testcontainer.ts` also needs to be updated

Reply:
Fixed in 88d7ca8.